lib/matplotlib/__init__.py

```diff
@@ -1690,7 +1690,7 @@ def param(func):
                 arg_names = []
             elif len(_arg_names) > 1 and (positional_parameter_names is None):
                 # we got no manual parameter names but more than an 'ax' ...
-                if len(set(replace_names) - set(_arg_names[1:])) == 0:
+                if len(replace_names - set(_arg_names[1:])) == 0:
```

Comment:
Not really sure I understand why this is needed and if its safe if replace_names contains duplicates.